kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java

```diff
@@ -102,6 +139,118 @@ public static class LogAppendInfo {
         private KopLogValidator.CompressionCodec targetCodec;
     }
 
+    @Data
+    @ToString
+    @Accessors(fluent = true)
+    @AllArgsConstructor
+    public static class ReadRecordsResult {
```

Comment:
I'm wondering if this recyclable class is meaningful? This object will finally be converted to `FetchResponse.PartitionData<Records>`. The only concern is the `DecodeResult` object that need to be recycled.

We can simply use a `Pair<FetchResponse.PartitionData<Records>, DecodeResult>`. Or you can insist the recyclable way to wrap these two classes.

Reply:
We also need `Position lastPosition;`, so I think we should keep it.